hathor/transaction/storage/transaction_storage.py

```diff
@@ -988,6 +998,58 @@ def _topological_fast(self) -> Iterator[BaseTransaction]:
         yield from cur_blocks
         yield from cur_txs
 
+    def _topological_fast2(self) -> Iterator[BaseTransaction]:
+        import heapq
+        from dataclasses import dataclass, field
+        cur_timestamp = settings.GENESIS_TIMESTAMP
+
+        @dataclass(order=True)
+        class Item:
+            timestamp: int
+            is_transaction: bool
+            tx: BaseTransaction = field(compare=False)
+
+            def __init__(self, tx: BaseTransaction):
+                self.timestamp = tx.timestamp
+                self.is_transaction = tx.is_transaction
+                self.tx = tx
+
+        to_visit: List[Item] = list(map(Item, self.get_all_genesis()))
+        heapq.heapify(to_visit)
+        visited: Set[bytes] = set()
+        to_check: List[Item] = []
+        to_yield: List[Item] = []
+        while (to_visit or to_check or to_yield):
+            while (to_visit and to_visit[0].timestamp <= cur_timestamp) or (not to_check):
```

Comment:
I'm almost sure we don't need this `cur_timestamp` to control the order. I guess it would simplify this implementation. Let me think more about it.

Reply:
We don't, it was trying to fix a problem that wouldn't need fixing if we just include the type of tx before.

Reply:
Ok!